pkg/providers/gcp/errors.go

```diff
@@ -0,0 +1,151 @@
+package gcp
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// googleRpcDetail is a minimal struct to map the objects inside the "Details" array.
+// We only parse the fields we specifically need, such as "@type" and "reason."
+type googleRpcDetail struct {
```

Comment:
Let's be more explicit, we can rename `google` instances to googleCloudProvider or gcp for short.